### ISSUE TITLE
Problems not being updated

### ISSUE THREAD

**Issue by anthonytec2** [microsoft/vscode-cpptools#6887]
Type: General
**Describe the bug**

I am having an issue when I first create an issue in my c++ code that causes a problem, which I then go back, solve and save. As a result, sometimes the problems area will not update this fix.  I have tried to debug this looking through the logs, I have found that I get both an error squiggle 1 and 0 count during saving. Would someone happen to know what I can do further to diagnosis why I am getting both 1 and 0 error squiggle counts? 

- OS and Version: macOs 10.15.7
- VS Code Version:1.52.1
- C/C++ Extension Version:v1.2.0-insiders3
- Other extensions you installed : None except c++ extension
- A clear and concise description of what the bug is: Problems pane is not being updated with an edit that fixes a syntax error. 

**To Reproduce** **Screenshots**
![image](https://user-images.githubusercontent.com/11021542/106545977-61b69000-64d8-11eb-9e00-84daadbf1675.png)

**Additional context**

[log.txt](https://github.com/microsoft/vscode-cpptools/files/5908316/log.txt)

**Comment by sean-mcmanus** [comment]
Does setting C_Cpp.intelliSenseCacheSize to 0 fix the issue? Otherwise, it seems like 2 IntelliSense processes are being created for the same source file...might be some issue related to pony.hpp. Are you able to get a repro with just a source file that doesn't have headers?

**Comment by anthonytec2** [comment]
I just tried setting intelliSenseCacheSize to 0 and this does not fix the issue. Just to show you exactly what I am referring to, I have attached a video of the problem with the corresponding log file. 


https://user-images.githubusercontent.com/11021542/106548781-a133ab00-64dd-11eb-9a74-03c593c0a905.mp4

[log2.txt](https://github.com/microsoft/vscode-cpptools/files/5908440/log2.txt)

**Comment by sean-mcmanus** [comment]
Are you able to repro it with a simple setup like

pony.cpp
```

#include <iostream>
#include "pony.h"

std::ostream &operator<<(std::ostream &os, const Pony &c)
{
    os << "dfdd" << c.getName() << std::endl;

     return os;
}
```
pony.h

```
class Pony
{
public:
    std::string getName() const { return ""; }
};

```

If not, are you able to determine what is triggering the repro in your case?

**Comment by anthonytec2** [comment]
I just tried the simple setup you suggested in a new directory. The issue does not reproduce in this new directory. I additionally, copied the original files to this directory and the issue did not reproduce. It seems to be something related to specially that file and directory. 

After a save, the issue still persists. Although, if I restart my editor, the issue goes away.

**Comment by sean-mcmanus** [comment]
Yeah, that is very odd. Your logs indicate that pony.hpp is somehow getting its own TU, which seems incorrect (it's probably supposed to share the TU with pony.cpp), so that could somehow be the root issue.

Can you try running the C/C++: Reset IntelliSense Database command? That holds some info on which source file to associate a header with, so something might be corrupt with that.

**Comment by anthonytec2** [comment]
Just got back to this, I began editing a new header file and new source file.Throughout the begining of my VScode session things are mostly stable and the problems field is being update. Although, at some point, things become unstable and I no longer still problems being updates and get many error squiggles throughout the code. 

I tried to reset the IntelliSense Database, and that does not help. 
[log3.txt](https://github.com/microsoft/vscode-cpptools/files/5915635/log3.txt)
![image](https://user-images.githubusercontent.com/11021542/106688367-f7ffba00-659b-11eb-9b36-bf5e8b97f22f.png)
